# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1341773010 25200
#      Sun Jul 08 11:43:30 2012 -0700
# Node ID 4563d99358532978d69b0abda3c91b97e63bc454
# Parent  bcf86cc2f1ee16c04a99fdd111dcd7f0131a28f7
maint: Allow scatter demo to be run under Octave or Matlab for plot comparisons

* __scatter__.m: Use columns() rather than size(c,2) for readability.

* scatter.m: Use single quotes in demo code so that it will run under Matlab.

diff --git a/scripts/plot/private/__scatter__.m b/scripts/plot/private/__scatter__.m
--- a/scripts/plot/private/__scatter__.m
+++ b/scripts/plot/private/__scatter__.m
@@ -263,17 +263,17 @@ function h = render_size_color(hg, vert,
   if (isscalar (s))
     x = vert(:,1);
     y = vert(:,2);
     z = vert(:,3:end);
     toolkit = get (ancestor (hg, "figure"), "__graphics_toolkit__");
     ## Does gnuplot only support triangles with different vertex colors ?
     ## TODO - Verify gnuplot can only support one color. If RGB triplets
     ##        can be assigned to each vertex, then fix __go_draw_axe__.m
-    gnuplot_hack = (numel (x) > 1 && size(c, 2) == 3
+    gnuplot_hack = (numel (x) > 1 && columns (c) == 3
                     && strcmp (toolkit, "gnuplot"));
     if (ischar (c) || ! isflat || gnuplot_hack)
       if (filled)
         h = __go_patch__ (hg, "xdata", x, "ydata", y, "zdata", z,
                           "faces", 1:numel(x), "vertices", vert,
                           "facecolor", "none", "edgecolor", "none",
                           "marker", marker,
                           "markeredgecolor", "none",
diff --git a/scripts/plot/scatter.m b/scripts/plot/scatter.m
--- a/scripts/plot/scatter.m
+++ b/scripts/plot/scatter.m
@@ -88,17 +88,17 @@ endfunction
 %! scatter (x, y, 'r');
 %! title ('Scatter plot with red bubbles');
 
 %!demo
 %! clf;
 %! x = randn (100, 1);
 %! y = randn (100, 1);
 %! c = x .* y;
-%! scatter (x, y, 20, c, "filled");
+%! scatter (x, y, 20, c, 'filled');
 %! title ('Scatter with colors');
 
 %!demo
 %! clf;
 %! x = randn (100, 1);
 %! y = randn (100, 1);
 %! scatter (x, y, [], sqrt (x.^2 + y.^2));
 %! title ('Scatter plot with bubble color determined by distance from origin');
